js/app/demo/app.jsx

```diff
@@ -15,7 +15,7 @@ define(['react', 'jsx!editor', 'jsx!messages', 'jsx!fixedCode', 'jsx!configurati
                 return result;
             }
 
-            return {
+            var initialState = JSON.parse(window.localStorage.getItem('linterDemoState')) || {
```

Comment:
This will result in the same issues that we had with the old behavior (i.e. every time we add a new rule, it doesn't show up in the demo, until you manually wipe out localStorage). We should either add version number, and invalidate localStorage on version change, or merge state from localStorage with new rules (might still result in bugs).